### ISSUE TITLE
Cannot get sorted and compressed in-line citations with natbib

### ISSUE THREAD

**Issue by hkaspersen** [rstudio/bookdown#781]
I am creating a pdf-book with `natbib` and `\setcitestyle{numbers,comma,square}` in my preamble. As of now, I see that the in-line citations doesn't compress, like this:

`[1,2,3,4,5]` instead of `[1 - 5]`

I tried the following:

- `\biboptions{sort&compress}` in preamble, this throws the error `undefined control sequence`
- `\setcitestyle{square,comma,numbers,sort&compress}` in preamble, this does not change the in-line citations at all, same with `\PassOptionsToPackage{square,comma,numbers,sort&compress}{natbib}` 
- `\setcitestyle{numbers,sort&compress}` alone, to see if something was messing it up

None of these options seem to change the in-line citations at all, so I asked on StackExchange, but only got an [answer relevant for LaTeX directly](https://tex.stackexchange.com/questions/510504/compressing-in-line-references-with-natbib-and-bookdown?noredirect=1#comment1290857_510504). I feel like it has something to do with the YAML in the index.Rmd? I am unsure how things work under the hood here, but I cannot seem to get it to work.

Reproducible example:

`index.Rmd`

```
--- 
title: "Test book"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
output:
  bookdown::pdf_book:
    latex_engine: xelatex
    fig_caption: yes
    citation_package: natbib
    includes:
      in_header: preamble.tex
bibliography: [library.bib, packages.bib]
link-citations: yes
---

# Test book

Testing compressed references [@Miski1983;@Blount2015;@OBrien2011].
```

`library.bib`:

```

@article{Miski1983,
  title = {Antibacterial Activity Studies of Flavonoids from {{Salvia}} Palaestina.},
  volume = {46},
  issn = {0163-3864},
  number = {6},
  journal = {Journal of natural products},
  url = {http://www.ncbi.nlm.nih.gov/pubmed/6677714},
  author = {Miski, M and Ulubelen, A and Johansson, C and Mabry, T J},
  year = {1983},
  pages = {874--5},
  pmid = {6677714}
}

@article{Blount2015,
  title = {The Unexhausted Potential of {{E}}. Coli},
  volume = {4},
  issn = {2050-084X},
  journal = {eLife},
  doi = {10.7554/eLife.05826},
  url = {https://elifesciences.org/articles/05826},
  author = {Blount, Zachary D.},
  month = mar,
  year = {2015},
  pages = {1--12},
  pmid = {25807083}
}

@article{OBrien2011,
  title = {Effect of Diet and Gut Dynamics on the Establishment and Persistence of {{Escherichia}} Coli},
  volume = {157},
  issn = {1350-0872},
  number = {5},
  journal = {Microbiology},
  doi = {10.1099/mic.0.047092-0},
  url = {http://mic.microbiologyresearch.org/content/journal/micro/10.1099/mic.0.047092-0},
  author = {O'Brien, Claire L. and Gordon, David M.},
  month = may,
  year = {2011},
  pages = {1375--1384}
}
```

`preamble.tex`
```
\usepackage{booktabs}

\setcitestyle{numbers,comma,square,sort&compress}
```

Which compiles into this:
![test](https://user-images.githubusercontent.com/35805211/66028482-82925e00-e4fd-11e9-8a2a-5cb844b0b2cb.png)

`Session info`

```
R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362), RStudio 1.2.1335

Locale:
  LC_COLLATE=Norwegian Bokmål_Norway.1252  LC_CTYPE=Norwegian Bokmål_Norway.1252    LC_MONETARY=Norwegian Bokmål_Norway.1252
  LC_NUMERIC=C                             LC_TIME=Norwegian Bokmål_Norway.1252    

Package version:
  base64enc_0.1.3 bookdown_0.13   digest_0.6.21   evaluate_0.14   glue_1.3.1      graphics_3.6.1  grDevices_3.6.1 highr_0.8      
  htmltools_0.3.6 jsonlite_1.6    knitr_1.25      magrittr_1.5    markdown_1.1    methods_3.6.1   mime_0.7        Rcpp_1.0.2     
  rmarkdown_1.15  stats_3.6.1     stringi_1.4.3   stringr_1.4.0   tinytex_0.16    tools_3.6.1     utils_3.6.1     xfun_0.9       
  yaml_2.2.0   
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.name/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('bookdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/bookdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by atusy** [comment]
I guess this is an issue of Pandoc because citations are supported by Pandoc rather than bookdown ( https://www.pandoc.org/MANUAL.html#citations ).
If you want to look further into the problem, I'd recommend specify `keep_md: true` and `keep_tex:true` in the YAML front matter and see the contents of intermidiate files.

**Comment by hkaspersen** [comment]
Thanks for your reply. I looked in the `.tex` file, and the only thing I could find was this:
`Testing compressed references \citep{Miski1983, Blount2015, OBrien2011}.`

I'm not very well versed with LaTeX, pandoc and natbib unfortunately. Do you have any idea what it may be?

**Comment by atusy** [comment]
Hmm... so tex file seems to be fine.

Did you try

```tex
\usepackage[sort&compress,numbers]{natbib}
```

as suggested in the the SO answer?

**Comment by hkaspersen** [comment]
Yes i tried that, it throws the error `LaTeX Error: Option clash for package natbib`

**Comment by atusy** [comment]
I guess it is because of `citation_package: natbib`.
What happens when removing `citation_package: natbib`, and specify only `\usepackage[sort&compress,numbers]{natbib}`?

**Comment by hkaspersen** [comment]
Still the same error

**Comment by atusy** [comment]
I checked the document and you have to specify `citation_package: none` to omit `natbib`.

**Comment by hkaspersen** [comment]
Then it seems like natbib is overridden, because the citations look like this:

`(Miski et al. 1983; Blount 2015; O’Brien and Gordon 2011)`

**Comment by atusy** [comment]
Pandoc's `natbib` option seems not to support extra arguments.
My new guess is that you have to use citations in the style of LaTeX rather than Pandoc's markdown.
However, I am not familiar with LaTeX...

**Comment by hkaspersen** [comment]
Do you mean as `\citep` instead of `[@]`?

**Comment by atusy** [comment]
Yes, and also to include bib files in the way of LaTeX.

**Comment by hkaspersen** [comment]
I see that a similar issue have been asked before: 
https://stackoverflow.com/questions/44770287/adding-natbib-options-for-pdf-output

That was two years ago. @yihui What is the status of this right now? Does the `natbiboptions` command work?

**Comment by hkaspersen** [comment]
It seems that `natbiboptions` work in the YAML like this: `natbiboptions: [numers, square, comma]`, but I cannot seem to add the option `sort&compress`, as the `&` mess things up

**Comment by hkaspersen** [comment]
Kind of fixed it by using `natbiboptions: "numbers,square,comma,sort,compress"` instead of `sort&compress`. I guess that'll do!
